engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -408,12 +407,12 @@ public void reset() {
    * @return a copy of this scope modulo the requested types
    */
   public ModuleScope withTypes(List<String> typeNames) {
-    Map<String, Object> polyglotSymbols = new HashMap<>(this.polyglotSymbols);
-    Map<String, Type> requestedTypes = new HashMap<>(this.types);
-    Map<Type, Map<String, Supplier<Function>>> methods = new ConcurrentHashMap<>();
```

Comment:
cc @hubertp. Note that I am getting rid of `ConcurrentHashMap` here and no tests are failing. This should be a proof that `ModuleScope` is intended to be single-threaded. Note that it would even not be possible to use it from more threads, as, e.g., in `reset` method in https://github.com/enso-org/enso/pull/9844/files#diff-c3869665e0ec34703f8e3eea0464924c0e4260eab392dbccdc202ec1f0270379L399, a normal `HashMap` is reassigned to the `methods` field instead of `ConcurrentHashMap`.